go/arrow/array/decimal128.go

```diff
@@ -229,6 +253,67 @@ func (b *Decimal128Builder) newData() (data *Data) {
 	return
 }
 
+func (b *Decimal128Builder) unmarshalOne(dec *json.Decoder) error {
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	var out *big.Float
+
+	switch v := t.(type) {
+	case float64:
+		out = big.NewFloat(v)
+	case string:
+		out, _, err = big.ParseFloat(v, 10, 0, big.ToNearestAway)
+		if err != nil {
+			return err
+		}
+	case json.Number:
```

Comment:
I might be misunderstanding but it seems like json.Number is an alias for string (https://pkg.go.dev/encoding/json#Number).  Should this go through the same logic above for string?

Reply:
It's not an alias but an actual type, which would only get used if the json encoder sets an option to use it. The `json.Number` type provides helper functions for parsing from the string, so the logic wouldn't be the same as using string, but instead using the helper functions directly and checking the error returns from them

Reply:
But it like accessing the [string literal](https://pkg.go.dev/encoding/json#Number.String) is supposed.  It seems like Float64 decoding loses precision needlessly

Reply:
good point, i've updated the `json.Number` case to work the same as the `string` case and fixed the precision to be 128